### PR TITLE
Handle multiple resource for same url - fix #48

### LINKED ISSUE #48: Handle multiple resources for one url case
Multiple resources can point to the same URL. This is currently not handled by `udata-piwik` when counting downloads, cf:
https://sentry.data.public.lu/share/issue/352e37353032/.

### PR DESCRIPTION
**Pull request by abulte**
This PR:

- Refactors the `counter` logic by handling downloads in a separate class
- Handles the case of multiple resources with the same URL. Unfortunately we can not detect exactly what resource has been hit (in the case of `hashed_url` detection) so there will be some duplicate metrics across resources.

### CODE REVIEW

File: udata_piwik/metrics.py [hunk 1]
```diff
@@ -202,3 +205,46 @@ def aggregate_reuses_daily(org, day):
...
+                dcommands.update({
+                    'set__resources__{0}__metrics__{1}'.format(i, k): 0
+                    for k in KEYS
+                })
+            try:
+                dataset.update(**dcommands)
+            except Exception:
+                log.exception('Unable to clear dataset %s', dataset.id)
+
+    commands = dict(('unset__values__{0}'.format(k), '1') for k in KEYS)
```
Comment (noirbizarre) [inline udata_piwik/metrics.py]: Quotes around `1` can be safely removed, Mongo just expect a truthy value

File: udata_piwik/utils.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        raise RuntimeError('Attempted to match a URL without the '
+                           'application context being pushed. This has to be '
+                           'executed when application context is available.')
+
+    url_adapter = appctx.url_adapter
+    if url_adapter is None:
+        raise RuntimeError('Application was not able to create a URL '
+                           'adapter for request independent URL matching. '
+                           'You might be able to fix this by setting '
+                           'the SERVER_NAME config variable.')
```
Comment (noirbizarre) [inline udata_piwik/utils.py]: :ok_hand:

Comment (abulte) [reply udata_piwik/utils.py]: your code ;-)

Comment (noirbizarre) [reply udata_piwik/utils.py]: :joy: Self congrat so ! :joy:

**Pull request merged**